packages/mdc-textfield/foundation.js

```diff
@@ -190,6 +194,27 @@ class MDCTextfieldFoundation extends MDCFoundation {
   }
 
   /**
+   * Changes the floating label's state
+   * @private
+   */
+  updateLabelFloat_() {
+    const input = this.getNativeInput_();
+    const {LABEL_FLOAT_ABOVE} = MDCTextfieldFoundation.cssClasses;
+
+    if (this.isFocused_ || input.value) {
+      if (!this.labelFloatIsActive_) {
+        this.labelFloatIsActive_ = true;
```

Comment:
instead of you using this.labelFloatIsActive as a internal state variable to keep track of "if the LABEL_FLOAT_ABOVE class has been added". Why not add a method to the adapter to check `adapter_.labelHasClass(LABEL_FLOAT_ABOVE)`?